### DISCUSSION TITLE
Change send method to chunkedSender throwing error

### DISCUSSION THREAD

**Discussion by vinodkhandelwal** [rpldy/react-uploady#188] [Q&A] [answered]
Hi,

I am trying to use Uploader for chunked upload and changing send method to Chunked sender. Sample code below - 

import createUploader, { UPLOADER_EVENTS } from "@rpldy/uploader";
import {createChunkedSender} from "@rpldy/chunked-sender";

 const uploader = createUploader({
      autoUpload: false,
      grouped: true,
      destination: { url: "http://localhost:3001" },
      params:{"name": fields},
      send: createChunkedSender({}).send,      
    });

on executing i am getting error.  - 

Uncaught (in promise) TypeError: trigger is not a function
    at triggerUpdater.js:5
    at new Promise (<anonymous>)
    at doTrigger (triggerUpdater.js:4)
    at push../node_modules/@rpldy/shared/lib/esm/triggerUpdater.js

I am not getting what is wrong that i am doing.

**Comment by yoavniran** [comment, marked as answer]
Hey @vinodkhandelwal 
The easiest way to use chunked sender is through its enhancer:


``` javascript

import getChunkedEnhancer from "@rpldy/chunked-sender";

export interface ChunkedOptions {
    chunked?: boolean;
    chunkSize?: number;
    retries?: number;
    parallel?: number;
}


const myChunkedOptions: ChunkedOptions = {
  //see interface above
};

 const chunkedEnhancer = getChunkedEnhancer(myChunkedOptions);

const uploader = createUploader({
   autoUpload: false,
   grouped: true,
   destination: { url: "http://localhost:3001" },
   params:{"name": fields},
   enhancer: chunkedEnhancer
});

```

Ill move this to the discussions area so it can be useful for others in the future.

Let me know if it works for you with the suggested code

**Reply by vinodkhandelwal** [reply]
Thanks a lot @yoavniran. It worked perfectly.
